pkg/roachprod/testselector/sfselector/selector.go

```diff
@@ -0,0 +1,307 @@
+			selectedTestCount++
+		}
+	}
+	if req.SelectFromSuccessPct > 0 && len(skippedTests) > 0 {
+		// need to select some tests from th success list (skipped tests)
+		numberOfTestsToSelect := len(skippedTests) * req.SelectFromSuccessPct / 100
+		// the tests are sorted by the number of runs. So, simply iterate over teh list
+		// and select the first count of "numberOfTestsToSelect"
+		for i := range skippedTests {
+			if i == numberOfTestsToSelect {
+				break
+			}
+			// mark the skipped tests as selected="yes"
+			skippedTests[i][1] = "yes"
+			selectedTestCount++
+		}
+	}
+	// add all the skipped tests (some may have been selected based on selectFromSuccessPct)
+	td.testInfos = append(td.testInfos, skippedTests...)
+	fmt.Printf("%d out of %d tests selected!\n", selectedTestCount, len(td.testInfos))
+	return td, nil
+}
+
+// getConnect makes connection to snowflake and returns the connection.
+func getConnect(ctx context.Context) (*gosql.DB, error) {
+	username, password, err := getSFCreds(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	dsn, err := sf.DSN(&sf.Config{
+		Account:   account,
+		Database:  database,
+		Schema:    schema,
+		Warehouse: warehouse,
+		Password:  password,
+		User:      username,
+	})
+	if err != nil {
+		return nil, err
+	}
+	db, err := gosql.Open("snowflake", dsn)
+	if err != nil {
+		return nil, err
+	}
+	return db, nil
+}
+
+// getCSVInMemory creates a CSV file in memory from a testDetails
+func getCSVInMemory(td *testDetails) (*bytes.Buffer, error) {
+	buffer := &bytes.Buffer{}
+	w := csv.NewWriter(buffer)
+	defer w.Flush()
+	if err := w.Write(td.columnHeaders); err != nil {
+		return nil, err
+	}
+	if err := w.WriteAll(td.testInfos); err != nil {
+		return nil, err
+	}
+	return buffer, nil
+}
+
+// uploadCSVToGCS uploads a CSV file in memory to Google Cloud Storage
+func uploadCSVToGCS(csvBuffer *bytes.Buffer, cloud, suite string) error {
+	ctx := context.Background()
+	client, err := storage.NewClient(ctx, option.WithScopes(storage.ScopeFullControl))
+	if err != nil {
+		return fmt.Errorf("failed to create GCS client: %w", err)
+	}
+	defer func() { _ = client.Close() }()
+
+	wc := client.Bucket(bucket).Object(
+		fmt.Sprintf("%s-%s-%s.%s", testsFileLocation, suite, cloud, testsCsvExtension)).NewWriter(ctx)
+	if _, err = io.Copy(wc, csvBuffer); err != nil {
+		return fmt.Errorf("failed to write CSV to GCS: %w", err)
+	}
+	if err := wc.Close(); err != nil {
+		return fmt.Errorf("failed to close GCS writer: %w", err)
+	}
+
+	return nil
+}
+
+// getSFCreds gets the snowflake credentials from the secrets manager
+func getSFCreds(ctx context.Context) (string, string, error) {
+	client, err := secretmanager.NewClient(ctx)
+	if err != nil {
+		return "", "", fmt.Errorf("failed to setup secrets manager client: %w", err)
+	}
+	defer func() { _ = client.Close() }()
+	// fetch the secret from the secrets manager
+	result, err := client.AccessSecretVersion(ctx, &secretmanagerpb.AccessSecretVersionRequest{
```

Comment:
Let's use an environment variable for this instead of secret manager. Dev-inf can guide you through setting the environment variable up when we get there.